client/a8c-for-agencies/sections/signup/signup-v2/components/multi-step-form/contact-form/hooks/use-contact-form-validation.ts

```diff
@@ -0,0 +1,81 @@
+import emailValidator from 'email-validator';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useState } from 'react';
+import { isSiteActive } from 'calypso/a8c-for-agencies/components/form/utils';
+import { AgencyDetailsSignupPayload } from 'calypso/a8c-for-agencies/sections/signup/types';
+import { preventWidows } from 'calypso/lib/formatting';
+
+export const CAPTURE_URL_RGX =
+	/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.][a-z0-9]+)*\.[a-z]{2,63}(:[0-9]{1,5})?(\/.*)?$/i;
+const CAPTURE_SOCIAL_URL_RGX =
+	/^(https?:\/\/)?(www\.)?(facebook\.com|linkedin\.com|instagram\.com)(\/.*)?$/i;
+
+type ValidationState = {
+	firstName?: string;
+	lastName?: string;
+	agencyName?: string;
+	agencyUrl?: string;
+	email?: string;
+};
+
+const useContactFormValidation = () => {
+	const translate = useTranslate();
+	const [ validationError, setValidationError ] = useState< ValidationState >( {} );
+	const [ isValidating, setIsValidating ] = useState( false );
+
+	const updateValidationError = ( newState: ValidationState ) => {
+		return setValidationError( ( prev ) => ( { ...prev, ...newState } ) );
+	};
+
+	const validate = useCallback(
+		async ( payload: Partial< AgencyDetailsSignupPayload > ) => {
+			const newValidationError: ValidationState = {};
+			setIsValidating( true );
+			if ( payload.firstName === '' ) {
```

Comment:
What do you think about adding `.trim()` here to ensure we won't allow whitespaces?